### PR TITLE
Reduce integration test flakiness

### PR DESCRIPTION
**Pull request by gdbelvin**
This PR removes the dependency on timeouts to advance the test which are tricky to time correctly and vary from platform to platform.  

It also removes client retry behavior during race conditions between clients. This prevents two clients from getting in an eternal race condition fight.

### CODE REVIEW

File: core/client/client.go [hunk 1]
```diff
@@ -228,15 +228,14 @@ func (m uint64Slice) Len() int           { return len(m) }
 func (m uint64Slice) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
 func (m uint64Slice) Less(i, j int) bool { return m[i] < m[j] }
 
-// Update creates an UpdateEntryRequest for a user,
-// attempt to submit it multiple times depending until ctx times out.
-// Returns context.DeadlineExceeded if ctx times out.
+// Update creates and submits a mutation for a user, and wait for it to appear.
```
[the remark this answers is not in the document]
Comment (gdbelvin) [reply core/client/client.go]: Done

File: core/client/client.go [hunk 2]
```diff
@@ -321,8 +303,9 @@ func (c *Client) WaitForUserUpdate(ctx context.Context, m *entry.Mutation) (*ent
 		case err == ErrWait:
 			// Try again.
 		case status.Code(err) == codes.DeadlineExceeded:
-			// Sometimes the timeout occurs during an rpc.
-			// Convert to a standard context.DeadlineExceeded for consistent error handling.
+			// Sometimes the timeout occurs during an rpc. Convert
```
[the remark this answers is not in the document]
Comment (gdbelvin) [reply core/client/client.go]: Wrapped to 95.

File: core/integration/monitor_tests.go [hunk 3]
```diff
@@ -102,8 +102,12 @@ func TestMonitor(ctx context.Context, env *Env, t *testing.T) {
 			opts := env.CallOpts(u.UserId)
 			cctx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
 			defer cancel()
-			if _, err = env.Client.Update(cctx, u, e.signers, opts...); err != context.DeadlineExceeded {
-				t.Fatalf("Could not send update request: %v", err)
+			m, err := env.Client.CreateMutation(cctx, u)
+			if err != nil {
+				t.Fatalf("CreateMutation(%v): %v", u.UserId, err)
+			}
+			if err := env.Client.QueueMutation(cctx, m, e.signers, opts...); err != nil {
```
[the remark this answers is not in the document]
Comment (gdbelvin) [reply core/integration/monitor_tests.go]: This is equivalent. From my reading, both `CreateMutation` and `QueueMutation` are using a timeout.

[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply core/integration/monitor_tests.go]: Yes, they do use the timeout. What looked confusing to me is that they don't return `context.DeadlineExceeded`, unlike in the old code.

[the remark this answers is not in the document]
Comment (gdbelvin) [reply core/integration/monitor_tests.go]: That's right.  The purpose of this refactor is to remove the use of `context.DeadlineExceeded` in tests.  

Previously the flow was:

- Update (QueueMutaiton and WaitForInclusion)
- Timeout
- Sequence
- Check for inclusion 

Now the flow is 
- QueueMutation
- Sequence
- Check for inclusion

**Pull request merged**